### ISSUE TITLE
leap-mx does not start in pipeline

### ISSUE THREAD

**Issue by cz8s** [pixelated/pixelated-platform#122]
our platform builds fail because leap-mx fails to start
```
root@pixelated:/home/leap/configuration# /etc/init.d/leap-mx start
Starting leap_mx: twistdUnhandled Error
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/twisted/application/app.py", line 642, in run
    runApp(config)
  File "/usr/lib/python2.7/dist-packages/twisted/scripts/twistd.py", line 23, in runApp
    _SomeApplicationRunner(config).run()
  File "/usr/lib/python2.7/dist-packages/twisted/application/app.py", line 376, in run
    self.application = self.createOrGetApplication()
  File "/usr/lib/python2.7/dist-packages/twisted/application/app.py", line 441, in createOrGetApplication
    application = getApplication(self.config, passphrase)
--- <exception caught here> ---
  File "/usr/lib/python2.7/dist-packages/twisted/application/app.py", line 452, in getApplication
    application = service.loadApplication(filename, style, passphrase)
  File "/usr/lib/python2.7/dist-packages/twisted/application/service.py", line 405, in loadApplication
    application = sob.loadValueFromFile(filename, 'application', passphrase)
  File "/usr/lib/python2.7/dist-packages/twisted/persisted/sob.py", line 210, in loadValueFromFile
    exec fileObj in d, d
  File "/usr/share/app/leap_mx.tac", line 24, in <module>
    from leap.mx.mail_receiver import MailReceiver
  File "/usr/lib/python2.7/dist-packages/leap/mx/mail_receiver.py", line 56, in <module>
    from leap.keymanager import openpgp
  File "/usr/lib/python2.7/dist-packages/leap/keymanager/__init__.py", line 64, in <module>
    from leap.common.events import emit_async, catalog
exceptions.ImportError: cannot import name emit_async

Failed to load application: cannot import name emit_async
```

**Comment by varac** [comment]
as a possible workaround, the leap keymanager package works:

```
root@pixelated:~# grep emit /usr/lib/python2.7/dist-packages/leap/keymanager/__init__.py
from leap.common.events import emit_async, catalog
            emit_async(catalog.KEYMANAGER_DONE_UPLOADING_KEYS, self._address)
        emit_async(catalog.KEYMANAGER_LOOKING_FOR_KEY, address)
            emit_async(catalog.KEYMANAGER_KEY_FOUND, address)
            emit_async(catalog.KEYMANAGER_KEY_NOT_FOUND, address)
            emit_async(catalog.KEYMANAGER_LOOKING_FOR_KEY, address)
            emit_async(
        emit_async(catalog.KEYMANAGER_STARTED_KEY_GENERATION, self._address)
root@pixelated:~# apt-cache policy keymanager
N: Unable to locate package keymanager
root@pixelated:~# apt-cache policy leap-keymanager
leap-keymanager:
  Installed: 0.4.1~48.gbp056ac2
  Candidate: 0.4.1~48.gbp056ac2
  Package pin: 0.4.1~48.gbp056ac2
  Version table:
 *** 0.4.1~48.gbp056ac2 999
        500 http://packages.pixelated-project.org/debian/ wheezy-snapshots/main amd64 Packages
        100 /var/lib/dpkg/status
     0.4.0+0~20150608200533.30+wheezy~1.gbp871360 999
        999 http://deb.leap.se/0.8/ wheezy/main amd64 Packages
root@pixelated:~# apt-get install leap-keymanager=0.4.0+0~20150608200533.30+wheezy~1.gbp871360
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following packages were automatically installed and are no longer required:
  libdrm-intel1 libdrm-nouveau1a libdrm-radeon1 libdrm2 libfontenc1 libgl1-mesa-dri libgl1-mesa-glx libglapi-mesa libgssglue1 libnfsidmap2 libpciaccess0 libtalloc2 libtdb1 libtirpc1
  libutempter0 libwbclient0 libx11-xcb1 libxcb-glx0 libxcb-shape0 libxcomposite1 libxdamage1 libxext6 libxfixes3 libxi6 libxinerama1 libxmuu1 libxrandr2 libxv1 libxxf86dga1 libxxf86vm1
  xbitmaps
Use 'apt-get autoremove' to remove them.
The following packages will be DOWNGRADED:
  leap-keymanager
0 upgraded, 0 newly installed, 1 downgraded, 0 to remove and 0 not upgraded.
Need to get 23.0 kB of archives.
After this operation, 31.7 kB of additional disk space will be used.
Do you want to continue [Y/n]? 
Get:1 http://deb.leap.se/0.8/ wheezy/main leap-keymanager all 0.4.0+0~20150608200533.30+wheezy~1.gbp871360 [23.0 kB]
Fetched 23.0 kB in 0s (32.6 kB/s)    
[master 036749e] saving uncommitted changes in /etc prior to apt run
 Author: go <go@pixelated.try.pixelated-project.org>
 15 files changed, 68892 insertions(+), 64857 deletions(-)
 delete mode 100755 sv/bigcouch/log/main/@40000000561608e80c90f8ec.s
 delete mode 100755 sv/bigcouch/log/main/@4000000056160b6d2654204c.s
 delete mode 100755 sv/bigcouch/log/main/@4000000056160dc902bc3fac.s
 delete mode 100755 sv/bigcouch/log/main/@4000000056161039223174c4.s
 delete mode 100755 sv/bigcouch/log/main/@40000000561612aa116ccae4.s
 delete mode 100755 sv/bigcouch/log/main/@40000000561614e4319cbcac.s
 create mode 100755 sv/bigcouch/log/main/@400000005616213e1b52c61c.s
 create mode 100755 sv/bigcouch/log/main/@40000000561623c03832778c.s
 create mode 100755 sv/bigcouch/log/main/@400000005616261c166200dc.s
 create mode 100755 sv/bigcouch/log/main/@40000000561628930a056824.s
 create mode 100755 sv/bigcouch/log/main/@4000000056162b0016da8c3c.s
 create mode 100755 sv/bigcouch/log/main/@4000000056162d5e3192355c.s
 rewrite sv/bigcouch/log/main/current (90%)
dpkg: warning: downgrading leap-keymanager from 0.4.1~48.gbp056ac2 to 0.4.0+0~20150608200533.30+wheezy~1.gbp871360
(Reading database ... 75553 files and directories currently installed.)
Preparing to replace leap-keymanager 0.4.1~48.gbp056ac2 (using .../leap-keymanager_0.4.0+0~20150608200533.30+wheezy~1.gbp871360_all.deb) ...
Unpacking replacement leap-keymanager ...
Setting up leap-keymanager (0.4.0+0~20150608200533.30+wheezy~1.gbp871360) ...
[master 909c53d] committing changes in /etc after apt run
 Author: go <go@pixelated.try.pixelated-project.org>
 1 file changed, 48 insertions(+)
root@pixelated:~# /etc/init.d/leap-mx start
Starting leap_mx: twistd.
root@pixelated:~# ps aux|grep mx
root     20144  0.7  0.6 313544 24452 ?        Sl   10:54   0:00 /usr/bin/python /usr/bin/twistd --pidfile=/var/run/leap_mx.pid --rundir=/var/lib/leap_mx/ --python=/usr/share/app/leap_mx.tac --logfile=/var/log/leap/mx.log --prefix=leap-mx
root     20163  0.0  0.0   7832   844 pts/1    S+   10:54   0:00 grep mx

```

**Comment by cz8s** [comment]
changed the buildscripts to force install leap-keymanager from leap. 
See pixelated-platform@90dc287c51ef92927e445dfc69d8e42eac685e94

**Comment by cz8s** [comment]
problem was, that python-leap-pycommon was still old. 
Installing pycommon and keymanager from our repos fixes the problem.

**Comment by cz8s** [comment]
fixed in pixelated/pixelated-platform@69a117b827ee81a58b94264fce996551dd7fc548
waiting for new build ...
